lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/__tests__/NxCollapsibleMultiSelect.test.tsx

```diff
@@ -4,355 +4,559 @@
+      it('has labeled text as specified in name of the option', function () {
+        const view = quickRender(),
+            options = view.getAllByRole('menuitemcheckbox');
+
+        expect(options).toHaveLength(4);
+        expect(options[0]).toHaveAccessibleName('all/none');
+        expect(options[1]).toHaveAccessibleName('Foo');
+        expect(options[2]).toHaveAccessibleName('Bar');
+        expect(options[3]).toHaveAccessibleName('Null');
       });
-      const toggleAllOptionWrapper = shallowRender.renderProp('renderToggleAllOption')();
 
-      expect(toggleAllOptionWrapper).toHaveProp('isChecked', false);
-    });
+      it('renders checked option when selected', function () {
+        const view = quickRender({
+              selectedIds: new Set(['foo'])
+            }),
+            options = view.getAllByRole('menuitemcheckbox');
 
-    it('renders unchecked <NxCheckbox> if not all options are selected', function () {
-      const shallowRender = getShallow({
-        selectedIds: new Set(['foo'])
+        expect(options[0]).not.toBeChecked();
+        expect(options[1]).toBeChecked();
+        expect(options[2]).not.toBeChecked();
+        expect(options[3]).not.toBeChecked();
       });
-      const toggleAllOptionWrapper = shallowRender.renderProp('renderToggleAllOption')();
 
-      expect(toggleAllOptionWrapper).toHaveProp('isChecked', false);
-    });
+      it('renders all unchecked options if selectedIds is not provided', function () {
+        const view = quickRender(),
+            options = view.getAllByRole('menuitemcheckbox');
 
-    it('renders checked <NxCheckbox> if all options are selected', function () {
-      const shallowRender = getShallow({
-        selectedIds: new Set(['foo', 'bar', null])
+        expect(options[0]).not.toBeChecked();
+        expect(options[1]).not.toBeChecked();
+        expect(options[2]).not.toBeChecked();
+        expect(options[3]).not.toBeChecked();
       });
-      const toggleAllOptionWrapper = shallowRender.renderProp('renderToggleAllOption')();
 
-      expect(toggleAllOptionWrapper).toHaveProp('isChecked', true);
-    });
+      it('calls component\'s onChange callback when an option is toggled', async function() {
+        const user = userEvent.setup(),
+            onChange = jest.fn(),
+            view = quickRender({ onChange }),
+            options = view.getAllByRole('menuitemcheckbox');
+
+        expect(onChange).not.toHaveBeenCalled();
+
+        await user.click(options[1]);
+
+        expect(onChange).toHaveBeenCalledWith(new Set(['foo']), 'foo');
+
+        await user.click(options[2]);
+
+        expect(onChange).toHaveBeenCalledWith(new Set(['bar']), 'bar');
+
+        await user.click(options[1]);
 
-    it('renders disabled <NxCheckbox> if component is disabled', function () {
-      const shallowRender = getShallow({
-        disabled: true
+        expect(onChange).toHaveBeenCalledWith(new Set(['foo']), 'foo');
       });
```

Comment:
Test having one or more already selected so we know the set containing the previous and new selections together gets constructed properly